src/tools/qt-gui/src/treeviewmodel.cpp

```diff
@@ -641,12 +618,50 @@ void TreeViewModel::synchronize ()
 		printKeys (ours, ours, ours);
 #endif
 
-		ThreeWayMerge merger;
-		AutoMergeConfiguration configuration;
-		configuration.configureMerger (merger);
+		try
+		{
+			// write our config
+			kdb.set (ours, m_root);
 
-		// write our config
-		m_kdb->synchronize (ours, m_root, merger);
+			// update our config (if no conflict)
+			kdb.get (ours, m_root);
+		}
+		catch (KDBException const &)
+		{
+			cout << "Bad" << endl;
+			// a conflict occurred, see if we can solve it with the merger
+
+			// refresh the key database
+			KeySet theirs = ours.dup ();
+			kdb.get (theirs, m_root);
+
+			// try to merge
+			ckdb::Key * informationKey = ckdb::keyNew (0, KEY_END);
+			ckdb::KeySet * c_merge_result = ckdb::elektraMerge (ours.getKeySet (), m_root.getKey (), theirs.getKeySet (),
+									    m_root.getKey (), modelBase.getKeySet (), m_root.getKey (),
+									    m_root.getKey (), ckdb::MERGE_STRATEGY_OUR, informationKey);
+			int numberOfConflicts = ckdb::getConflicts (informationKey);
+			ckdb::keyDel (informationKey);
+			if (c_merge_result != NULL)
+			{
+				// hurray, we solved the issue
+				kdb::KeySet result = c_merge_result;
+				kdb.set (result, m_root);
+				modelBase = result;
+			}
+			else
+			{
+				if (numberOfConflicts > 0)
+				{
+					emit showMessage (tr ("Error"), tr ("Could not merge keys due to merge conflicts."), "");
```

Comment:
Which keys were the problematic ones?

Reply:
elektraMerge does only keep track of the amount of conflicts, not about the exact keys.

Reply:
Would be good if the keys were somehow marked (e.g. by metadata). UIs are interested in which keys had the conflict.

Reply:
Do you want this in this PR already? Otherwise I'd open a new issue + PR for this. Then we could discuss if we use a new key set for this as the C++ merge did or the existing three key sets, for example.

Reply:
Yes, it can be in a separate issue + PR. (and is even better this way)

Reply:
See #3338